base/backend.Service.yaml

```diff
@@ -6,6 +6,7 @@ metadata:
       the same node if possible.
   labels:
     deploy: sourcegraph
+    sourcegraph-rbac-admin: non-escalated
```

Comment:
Sorry to put this in last-minute: should we reconsider the name and values too? As I understand it they are:

- `sourcegraph-rbac-admin: escalated`
- `sourcegraph-rbac-admin: non-escalated`

But.. what do these really mean? What is "escalated role based access control administration"?

Reply:
In other words, I wonder what we're describing here in practice and whether or not it would make sense to me when explained with docs and someone more experienced with k8s like @daxmc99 without docs.

Reply:
no problem, it's just search/replace :-)

there's two subsets of admins apparently: cluster-wide admins have permission to create/modify/delete cluster-wide resources (like ClusterRoles and ClusterRoleBindings). sometimes only these people can create a storage class for example. this group are the admins with super powers.

the other group are constrained to a namespace and can only create resources in that namespace.

this label marks things as needing the first group to create them or not. unfortunately you cannot mix a label inequality with -prune because all sorts of hell will break loose. so you have to name the non-special resources as not needing admin super powers and select them also with a label equality clause.

of course you can define even more granular and specific permissions with roles, but this is the most common.

Reply:
you can then ask the ones with super powers to do
```
./kubectl-apply-all.sh  -l sourcegraph-rbac-admin=escalated
```

and the other group can take over and do the

```
./kubectl-apply-all.sh  -l sourcegraph-rbac-admin=non-escalated
```

Reply:
so any name/value change is welcome (now that you know what they mean) :-)

Reply:
Got it, that is a very helpful explanation, thank you! In this case, I might propose:

```
sourcegraph-service-requires=admin
sourcegraph-service-requires=non-admin
````

This makes sense to me as an outsider to the norms of k8s, but does it make sense to you @uwedeportivo and others?

Reply:
it's not just service

how about

```
sourcegraph-resource-requires=cluster-admin
sourcegraph-resource-requires=no-cluster-admin
````

Reply:
That looks great to me!

Reply:
I'd like to have buy-in from 1 other on the team that this makes sense to them (Geoffrey, Dax, Dave, Beyang, etc)

Reply:
sure. i'll wait

Reply:
Not to jump on the bandwagon, but are these more appropriate as annotations instead of labels? Do we imagine k8s tooling to need to automatically select on these labels?

Reply:
yes, we do. we want to

```
./kubectl-apply-all.sh  -l sourcegraph-rbac-admin=escalated
```
and 
```
./kubectl-apply-all.sh  -l sourcegraph-rbac-admin=non-escalated
```

Reply:
+1 from me. This makes good sense delegating responsibilities between those with elevated admin privileges and those without.

Reply:
I'm a fan of the prefix. I think it helps communicate that these are specifically used by Sourcegraph.